pytools/graph.py

```diff
@@ -43,29 +49,46 @@
 Type Variables Used
 -------------------
 
-.. class:: T
+.. class:: NodeT
+
+    Type of a graph node, can be any type.
 
-    Any type.
+.. class:: GraphT
+
+    A :class:`collections.abc.Mapping` representing a directed
+    graph. The mapping contains one key representing each node in the
+    graph, and this key maps to a :class:`collections.abc.Collection` of its
+    successor nodes. Note that most functions expect that every graph node
+    is included as a key in the graph.
 """
 
 from typing import (Collection, TypeVar, Mapping, List, Optional, Any,
-                    Callable, Set, MutableSet, Dict, Iterator, Tuple, FrozenSet)
+                    Callable, Set, MutableSet, Dict, Iterator, Tuple, FrozenSet,
+                    Type)
+
+try:
+    from typing import TypeAlias
+except ImportError:
+    from typing_extensions import TypeAlias
+
+
+NodeT = TypeVar("NodeT")
```

Comment:
```suggestion
NodeT = TypeVar("NodeT", bound=Hashable)
```
? Not sure if that brings anything, but should be clearer since it's used as a key in the mapping.

Reply:
Good point, I added this in 4c8ad32